### PR TITLE
xdg-terminal-exec: init at unstable-2023-12-08

### PR DESCRIPTION
**Pull request by quantenzitrone**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/xd/xdg-terminal-exec/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}:
+stdenv.mkDerivation {
+  pname = "xdg-terminal-exec";
+  version = "unstable-20231208";
```
Comment (OPNA2608) [inline pkgs/by-name/xd/xdg-terminal-exec/package.nix:8]: https://github.com/NixOS/nixpkgs/blob/0838486e9183ce39398f01b221766e68b50f405c/pkgs/README.md?plain=1#L353

Date format is wrong. And the new convention sadly doesn't explain what to do when there is *no* latest upstream version <sub>( :confused: )</sub>, but inventing some version `0`[`.0`[`.0`]] seems to be the way to go.

(remember to adjust version in commit message as well)

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

**Comment by jtojnar** [comment]
Duplicate of https://github.com/NixOS/nixpkgs/pull/244886

**Comment by quantenzitrone** [comment]
> Duplicate of #244886

i should have looked

but the author of #244886 seems to be unresponsive

File: pkgs/by-name/xd/xdg-terminal-exec/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    rev = "04f37d4337b6ce157d4a7338dd600a32deb43a28";
+    hash = "sha256-QIPdF+/dMUEVcz5j9o+wQ4dnw2yWwz7slnLdMNETkGs=";
+  };
+
+  dontBuild = true;
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp xdg-terminal-exec $out/bin
```
Comment (eclairevoyant) [inline pkgs/by-name/xd/xdg-terminal-exec/package.nix:22]:
```suggestion
    install -Dm555 xdg-terminal-exec -t $out/bin
```
and the `mkdir` line can be removed

File: pkgs/by-name/xd/xdg-terminal-exec/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}:
+stdenv.mkDerivation {
```
Comment (eclairevoyant) [inline pkgs/by-name/xd/xdg-terminal-exec/package.nix:6]:
```suggestion
stdenvNoCC.mkDerivation {
```

File: pkgs/by-name/xd/xdg-terminal-exec/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  stdenv,
```
Comment (eclairevoyant) [inline pkgs/by-name/xd/xdg-terminal-exec/package.nix:3]:
```suggestion
  stdenvNoCC,
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

**Review by eclairevoyant — APPROVED** [review]

**Comment by onemoresuza** [comment]
# Reviewed points

Result of `nixpkgs-review pr 273386` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xdg-terminal-exec</li>
  </ul>
</details>

# Reviewed points

- [X] package path fits guidelines
- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package builds on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] `meta.description` is set and fits guidelines
- [X] `meta.license` fits upstream license
- [X] `meta.platforms` is set
- [X] `meta.maintainers` is set
- [X] build time only dependencies are declared in `nativeBuildInputs`
- [X] source is fetched using the appropriate function
- [X] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.

**Review by onemoresuza — APPROVED** [review]

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 273386` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xdg-terminal-exec</li>
  </ul>
</details>

The only non-posix executable i can see being called is `md5sum` from coreutils, which is also available in busybox.

Works with lxterminal, LGTM!

**Pull request merged**